includes/snmp.inc.php

```diff
@@ -370,140 +346,29 @@ function snmpwalk_cache_multi_oid($device, $oid, $array, $mib = null, $mibdir =
-        if (!strstr($value, 'at this OID') && isset($oid) && isset($first) && isset($second)) {
-            $index               = $first.'.'.$second.'.'.$third;
-            $array[$index][$oid] = $value;
-        }
-    }
-
-    return $array;
+    return snmp_output_to_array($data, $array, 3);
 }//end snmpwalk_cache_triple_oid()
 
-
-function snmpwalk_cache_twopart_oid($device, $oid, $array, $mib = 0)
+// TODO kill
+function snmpwalk_cache_twopart_oid($device, $oid, $array, $mib = null)
 {
-    $cmd = gen_snmpwalk_cmd($device, $oid, ' -OQUs', $mib);
-    $data = trim(external_exec($cmd));
-
-    foreach (explode("\n", $data) as $entry) {
-        list($oid,$value) = explode('=', $entry, 2);
-        $oid              = trim($oid);
-        $value            = trim($value);
-        $value            = str_replace('"', '', $value);
-        list($oid, $first, $second) = explode('.', $oid);
-        if (!strstr($value, 'at this OID') && isset($oid) && isset($first) && isset($second)) {
-            $array[$first][$second][$oid] = $value;
-        }
-    }
-
-    return $array;
+    return snmpwalk_cache_oid($device, $oid, $array, $mib);
 }//end snmpwalk_cache_twopart_oid()
 
-
-function snmpwalk_cache_threepart_oid($device, $oid, $array, $mib = 0)
+// TODO kill
+function snmpwalk_cache_threepart_oid($device, $oid, $array, $mib = null)
 {
-    global $debug;
-
-    $cmd = gen_snmpwalk_cmd($device, $oid, ' -OQUs', $mib);
-    $data = trim(external_exec($cmd));
-
-    foreach (explode("\n", $data) as $entry) {
-        list($oid,$value) = explode('=', $entry, 2);
-        $oid              = trim($oid);
-        $value            = trim($value);
-        $value            = str_replace('"', '', $value);
-        list($oid, $first, $second, $third) = explode('.', $oid);
-
-        if ($debug) {
-            echo "$entry || $oid || $first || $second || $third\n";
-        }
-
-        if (!strstr($value, 'at this OID') && isset($oid) && isset($first) && isset($second) && isset($third)) {
-            $array[$first][$second][$third][$oid] = $value;
-        }
-    }
-
-    return $array;
+    return snmpwalk_cache_oid($device, $oid, $array, $mib);
 }//end snmpwalk_cache_threepart_oid()
 
 
-function snmp_cache_slotport_oid($oid, $device, $array, $mib = 0)
-{
-    $cmd = gen_snmpwalk_cmd($device, $oid, ' -OQUs', $mib);
-    $data      = trim(external_exec($cmd));
-
-    foreach (explode("\n", $data) as $entry) {
-        $entry                  = str_replace($oid.'.', '', $entry);
-        list($slotport, $value) = explode('=', $entry, 2);
-        $slotport               = trim($slotport);
-        $value                  = trim($value);
-        if ($array[$slotport]['ifIndex']) {
-            $ifIndex               = $array[$slotport]['ifIndex'];
-            $array[$ifIndex][$oid] = $value;
-        }
-    }
-
-    return $array;
-}//end snmp_cache_slotport_oid()
-
-
-function snmp_cache_oid($oid, $device, $array, $mib = 0)
```

Comment:
This is used in `./includes/polling/test.inc.php` which whilst we don't actually tell people to use is something we should either bin off or make use of.